runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/FlinkBatchTransformTranslators.java

```diff
@@ -302,25 +301,8 @@ public void translateNode(Write.Bound<T> transform, FlinkBatchTranslationContext
     }
   }
 
-  private static class GroupByKeyOnlyTranslatorBatch<K, V> implements FlinkBatchPipelineTranslator.BatchTransformTranslator<GroupByKey.GroupByKeyOnly<K, V>> {
```

Comment:
`GroupByKey` expands not only into `GroupByKeyOnly` but also does the Windowing and timestamp assignment. In early Dataflow versions, this used to be different. When the changes came, we introduced an additional translator to skip the Window assignment. I would leave it as it is for now and do an immediate follow-up pull request where we get rid of this artifact. IMHO the GroupByKeyOnly translator should stay and GroupByKey should be removed.